cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/JSONParser.scala

```diff
@@ -56,6 +57,27 @@ object JSONParser {
     resolved
   }
 
+  def parseLimit(json: JsValue, option: QueryExeOption): (JsValue, QueryExeOption) = {
+    if (option.sliceMills <= 0) { // non-slicing query
+      (json, QueryExeOption(option.sliceMills, option.continueSeconds, None))
+    } else if (option.sliceMills > 0 && (json \ "batch").toOption.isEmpty) { // single slicing query
+      if ((json \\ "limit").isEmpty) {
+        (json, QueryExeOption(option.sliceMills, option.continueSeconds, None))
+      } else {
+        val limit = (json \ "select" \ "limit").as[Int]
+        val updatedSelectJson = (json \ "select").as[JsObject] ++ Json.obj("limit" -> Int.MaxValue)
```

Comment:
if we only `++`, the original `limit` property is still there, right?

Reply:
No, it will be overwritten. I had a unit test case in JsonParserTest to test it.

Reply:
Got it.

Reply:
in "parse limit in single query"